sys/kern/tty.c

```diff
@@ -555,6 +569,38 @@ static void tty_output(tty_t *tty, uint8_t c) {
     }
   }
   tty->t_column = col;
+  return true;
+}
+
+/*
+ * Write a single character to the terminal.
+ * If it can't immediately write the character due to lack of space
+ * in the output queue, it goes to sleep waiting for space to become available.
+ */
+static void tty_output_sleep(tty_t *tty, uint8_t c) {
+  while (!tty_output(tty, c)) {
+    tty_notify_out(tty);
+    if (tty->t_outq.count < TTY_OUT_LOW_WATER)
```

Comment:
What's the purpose of this line. At first glance I'd say this condition causes busy-looping...

Reply:
`tty_notify_out` may output characters synchronously. This line checks if it has output enough characters for us not to need to wait.

Reply:
Such comment here would be helpful while reading.

Reply:
Done.